src/qutip_tensorflow/core/data/tftensor.py

```diff
@@ -0,0 +1,98 @@
+import tensorflow as tf
+from tensorflow.errors import InvalidArgumentError
+import qutip
+import numbers
+
+__all__ = ["TfTensor"]
+
+
+class TfTensor(qutip.core.data.Data):
+    """This class provide a wraps around TensorFlow's Tensor. It will store data
+    as a Tensor of dtype `tensorflow.complex128`. Data will be expanded into a 2D
+    Tensor.
+
+    Parameters
+    ----------
+    data: array-like
+        Data to be stored. Accepts same array-like as `tensorflow.constant`.
+    shape: (int, int)
+        Shape of data. Default `None`, it tries to infer the shape from data
+        accessing the attribute `data.shape`.
+    copy: bool
+        If `True` (default) then the object is copied. Otherwise, copy will only
+        be made if tf.constant returns a copy of data(when input is not a
+        `Tenor`) or if a copy is needed to satisfy dtype (tensorflow.complex128) and
+        shape (2D Tensor)."""
+
+    def __init__(self, data, shape=None, copy=False):
```

Comment:
Is there any situation in which the backing data field of this object can be mutable?  I don't know tensorflow so well, but say for numpy
```python
>>> class Data:
...   def __init__(self, data):
...      self.data = np.array(data, copy=False)
...   def __repr__(self):
...     return f"Data({repr(self.data)})"
...
>>> x = np.ones((2,))
>>> y = Data(x)
>>> x
array([1., 1.])
>>> y
Data(array([1., 1.]))
>>> x *= 2
>>> x
array([2., 2.])
>>> y
Data(array([2., 2.]))
```
so the derived object has changed from under us.

If there is such a case, then the default argument to `copy` _must_ be `True`.  If not, then does the `copy` keyword argument serve any purpose?

Reply:
The above case would not work for tensorflow if using a `Tensor`:
```python
a = random_tensor((1,1)) # random tensor of shape (1,1)
b = TfTensor(a)
b._tf == a # True
a *= 2
b._tf == a #False
```
To my knowledge, there is no operation in a `tf.Tensor` that does not return a different tensor. `tf.Variables`, although not supported yet in qutip-tensorflow, do perform in-place operations with the `assign()` method. 

I can not think of a case where `copy=true` is necessary but it does have a different behaviour. For instance,
```python
a = tf.constant([[1+1j]])
b = TfTensor(a, copy=False) 
a is not b._tf # False
b = TfTensor(a, copy=True)
a is not b._tf # True 
``` 
Although I do not know when and why this could be useful (this may be why `tf.constant()` does not have a `copy` argument).

Reply:
If tensors are immutable, then the copy argument is pretty redundant. It's fine to have it for consistency, you may just want to mention that it does very little in the docstring.